src/ORM/Table.php

```diff
@@ -291,6 +297,9 @@ public function __construct(array $config = [])
         if (!empty($config['connection'])) {
             $this->setConnection($config['connection']);
         }
+        if (!empty($config['queryFactory'])) {
+            $this->queryFactory = $config['queryFactory'];
```

Comment:
This looks a bit like a chicken-and-egg situation. The factory constructor requires the table, and the table constructor "requires" the factory.

Reply:
Yes, perhaps the table instance should be passed as a method argument instance instead of the constructor?

Reply:
Also perhaps TableLocator should hold the factory instance(s) to avoid creating a factory instance per table.

Reply:
That would be a possibility. Maybe even the connection, given that a table's connection can be changed dynamically?

Reply:
> Also perhaps `TableLocator` should hold the factory instance(s) to avoid creating a factory instance per table.

I'd say what's important is that one could easily provide a factory for _all_ tables, so passing it to the table locator for all tables to receive could certainly be an option.

Reply:
One can now pass their custom query factory [here](https://github.com/cakephp/app/blob/d52e6ea5f9b8f728d71207a307ffa8e3dded8932/src/Application.php#L54) which would be used for all tables.

Reply:
Looks good, but could need a test or two for the global usage and the per alias config.

Reply:
Done